### PR TITLE
Added a brief explanation about Doctrine DBAL Session Storage

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | 2.3+ |
| Fixed tickets | #881 |

This PR fixes one of the oldest pending issues, dating back to Nov. 2011 :)

### CODE REVIEW

File: cookbook/configuration/pdo_session_storage.rst [hunk 1]
```diff
@@ -217,3 +217,79 @@ For MSSQL, the statement might look like the following:
...
+-----------------------------
+
+Symfony applications can also use a Doctrine DBAL based session storage. This
+alternative implementation is very similar to the ``PdoSessionHandler`` explained
+above, but uses a Doctrine connection and thus also works with non-PDO-based
+drivers like mysqli and OCI8.
+
+The only significant disadvantage of the Doctrine DBAL session storage comparing
+it with ``PdoSessionHandler`` is that you can only configure the name of the
+table used to store sessions, but not its column names.
```
Comment (cordoval) [inline cookbook/configuration/pdo_session_storage.rst]: i am not sure of the comma here

File: cookbook/configuration/pdo_session_storage.rst [hunk 2]
```diff
@@ -217,3 +217,79 @@ For MSSQL, the statement might look like the following:
...
+    .. code-block:: php
+
+        // app/config/config.php
+        use Symfony\Component\DependencyInjection\Definition;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        $container->loadFromExtension('framework', array(
+            ...,
+            'session' => array(
+                // ...,
```
Comment (cordoval) [inline cookbook/configuration/pdo_session_storage.rst]: are we doing that ...,? sounds weird

File: cookbook/configuration/pdo_session_storage.rst [hunk 3]
```diff
@@ -217,3 +217,79 @@ For MSSQL, the statement might look like the following:
...
+        </services>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        use Symfony\Component\DependencyInjection\Definition;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        $container->loadFromExtension('framework', array(
+            ...,
```
Comment (cordoval) [inline cookbook/configuration/pdo_session_storage.rst]: why arbitrarily doing // ...  and then here do ... without the // ?

**Comment by WouterJ** [comment]
There was already a PR on this: https://github.com/symfony/symfony-docs/pull/3914

We should pick the best things of both PRs and have one PR.

/cc @xocasdashdash

**Comment by javiereguiluz** [comment]
@WouterJ I'm sorry about that. Can we develop a bot that automatically applies the `has PR` label to the issues with a pending PR?

File: cookbook/configuration/pdo_session_storage.rst [hunk 4]
```diff
@@ -217,3 +217,85 @@ For MSSQL, the statement might look like the following:
...
+
+Doctrine DBAL Session Storage
+-----------------------------
+
+Symfony applications can also use a Doctrine DBAL based session storage. This
+alternative implementation is very similar to the ``PdoSessionHandler`` explained
+above, but uses a Doctrine connection and thus also works with non-PDO-based
+drivers like mysqli and OCI8.
+
+The only significant disadvantage of the Doctrine DBAL session storage comparing
```
Comment (stof) [inline cookbook/configuration/pdo_session_storage.rst]: note that these paragraphs are true for 2.3, but not for 2.6+ (the DBAL session storage has not been migrated to implement session locking yet AFAIK)

**Comment by javiereguiluz** [comment]
Closing it because there is another previous PR to fix this issue.

**Comment by javiereguiluz** [comment]
Reopening it because the other previous PR was closed precisely in favor of this PR.

**Comment by javiereguiluz** [comment]
What's left for this PR to be considered finished?

File: cookbook/configuration/pdo_session_storage.rst [hunk 5]
```diff
@@ -217,3 +217,85 @@ For MSSQL, the statement might look like the following:
...
+        $container->loadFromExtension('framework', array(
+            // ...
+            'session' => array(
+                // ...
+                'handler_id' => 'session.handler.dbal',
+            ),
+        ));
+
+        // ...
+        $container->setParameter('pdo.dbal_session_table', 'sess');
```
Comment (xabbuh) [inline cookbook/configuration/pdo_session_storage.rst]: The other examples don't have the `pdo.` prefix.

File: cookbook/configuration/pdo_session_storage.rst [hunk 6]
```diff
@@ -217,3 +217,85 @@ For MSSQL, the statement might look like the following:
...
+        </parameters>
+
+        <!-- app/config/services.xml -->
+        <services>
+            <!-- ... -->
+            <service id="session.handler.dbal" class="Symfony\Bridge\Doctrine\HttpFoundation\DbalSessionHandler">
+                <argument type="service" id="doctrine.dbal.default_connection" />
+                <argument>%dbal_session_table%</argument>
+            </service>
+        </services>
```
Comment (xabbuh) [inline cookbook/configuration/pdo_session_storage.rst]: I prefer to have a complete example:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<container xmlns="http://symfony.com/schema/dic/services"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:framework="http://symfony.com/schema/dic/symfony"
    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd
        http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">

    <framework:config>
        <!-- ... -->
        <framework:session handler-id="session.handler.dbal" />
    </framework:config>

    <parameters>
        <!-- ... -->
        <parameter key="dbal_session_table">session</parameter>
    </parameters>

    <!-- app/config/services.xml -->
    <services>
        <!-- ... -->
        <service id="session.handler.dbal" class="Symfony\Bridge\Doctrine\HttpFoundation\DbalSessionHandler">
            <argument type="service" id="doctrine.dbal.default_connection" />
            <argument>%dbal_session_table%</argument>
        </service>
    </services>
</container>
```

**Comment by weaverryan** [comment]
I actually dislike that we're giving people two equivalent options (or maybe this is not as good as the PDO option, since it doesn't have the new locking stuff). I'd like to *not* document this - since it doesn't add anything, it takes away.

Also, the biggest potential advantage - reusing the dbal connection - is not a good idea, according to the original PR: https://github.com/symfony/symfony/pull/2182#issuecomment-2104241

Anyone else agree?

**Comment by stof** [comment]
the DBAL implementation made sense in 2.3 because it allowed to lazy-load the connection. but the new 2.6+ PDO implementation also supports lazy-loading the PDO connection, so it is indeed less necessary.

**Comment by weaverryan** [comment]
Thanks Stof - I'm going to leave this as a "will not document"

**Pull request closed without merging**